### PR TITLE
OTLP metrics gRPC exporter

### PR DESCRIPTION
**Pull request by paivagustavo**
This is part of the #1827 todo list:
  - [x] add gRPC otlp trace exporter
  - [x] add gRPC otlp metrics exporter
  - [x] add HTTP otlp trace exporter
  - [ ] add HTTP otlp metrics exporter
  - [ ] refactor out the duplicated code (transform, connection, otlpconfig)
  - [ ] update examples  and remove old otlp gRPC/HTTP exporter
  - [ ] update docs

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Creates package `go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp` implementing a HTTP `otlptrace.Client` and offers convenience functions, `NewExportPipeline` and `InstallNewPipeline`, to setup and install a `otlptrace.Exporter` in tracing. (#1963)
 - Changes `go.opentelemetry.io/otel/sdk/resource.NewWithAttributes` to require a schema URL. The old function is still available as `resource.NewSchemaless`. This is a breaking change. (#1938)
 - Several builtin resource detectors now correctly populate the schema URL. (#1938)
+- Creates package `go.opentelemetry.io/otel/exporters/otlp/otlpmetric` that defines a metrics exporter that uses a `otlpmetric.Client` to send data.
+  Creates package `go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlptracegrpc` implementing a gRPC `otlpmetric.Client` and offers convenience functions, `NewExportPipeline` and `InstallNewPipeline`, to setup and install a `otlpmetric.Exporter` in metrics .(#TBD)
```
Comment (MrAlias) [inline CHANGELOG.md:44]:
```suggestion
  Creates package `go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlptracegrpc` implementing a gRPC `otlpmetric.Client` and offers convenience functions, `NewExportPipeline` and `InstallNewPipeline`, to setup and install a `otlpmetric.Exporter` in metrics .(#1991)
```

File: exporters/otlp/otlpmetric/exporter.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+func (e *Exporter) ExportKindFor(descriptor *metric.Descriptor, aggregatorKind aggregation.Kind) metricsdk.ExportKind {
+	return e.exportKindSelector.ExportKindFor(descriptor, aggregatorKind)
+}
+
+var _ metricsdk.Exporter = (*Exporter)(nil)
+
+// NewExporter constructs a new Exporter and starts it.
+func NewExporter(ctx context.Context, client Client, opts ...ExporterOption) (*Exporter, error) {
```
Comment (MrAlias) [inline exporters/otlp/otlpmetric/exporter.go:103]:
```suggestion
// New constructs a new Exporter and starts it.
func New(ctx context.Context, client Client, opts ...ExporterOption) (*Exporter, error) {
```

File: exporters/otlp/otlpmetric/exporter.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	exp := NewUnstartedExporter(client, opts...)
+	if err := exp.Start(ctx); err != nil {
+		return nil, err
+	}
+	return exp, nil
+}
+
+// NewUnstartedExporter constructs a new Exporter and does not start it.
+func NewUnstartedExporter(client Client, opts ...ExporterOption) *Exporter {
```
Comment (MrAlias) [inline exporters/otlp/otlpmetric/exporter.go:112]:
```suggestion
// NewUnstarted constructs a new Exporter and does not start it.
func NewUnstarted(client Client, opts ...ExporterOption) *Exporter {
```

File: exporters/otlp/otlpmetric/exporter.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+		return nil, nil, err
+	}
+
+	err = cntr.Start(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return exp, cntr, err
+}
```
Comment (MrAlias) [inline exporters/otlp/otlpmetric/exporter.go:164]:
```suggestion
```

File: exporters/otlp/otlpmetric/go.mod [hunk 5]
```diff
@@ -0,0 +1,76 @@
+module go.opentelemetry.io/otel/exporters/otlp/otlpmetric
+
+go 1.14
```
Comment (MrAlias) [inline exporters/otlp/otlpmetric/go.mod:3]:
```suggestion
go 1.15
```

File: exporters/otlp/otlpmetric/go.mod [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+replace go.opentelemetry.io/otel/metric => ../../../metric
+
+replace go.opentelemetry.io/otel/oteltest => ../../../oteltest
+
+replace go.opentelemetry.io/otel/trace => ../../../trace
+
+replace go.opentelemetry.io/otel/sdk/export/metric => ../../../sdk/export/metric
+
+replace go.opentelemetry.io/otel/sdk/metric => ../../../sdk/metric
```
Comment (MrAlias) [inline exporters/otlp/otlpmetric/go.mod:21]: Can we move these below the `required` section to remain consistent? Maybe just remove these and re-run the `crosslink` util?

File: exporters/otlp/otlpmetric/options.go [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlpmetric
+
+import metricsdk "go.opentelemetry.io/otel/sdk/export/metric"
+
+// ExporterOption are setting options passed to an Exporter on creation.
+type ExporterOption func(*config)
```
Comment (MrAlias) [inline exporters/otlp/otlpmetric/options.go:20]: This doesn't match the project configuration pattern: https://github.com/open-telemetry/opentelemetry-go/blob/main/CONTRIBUTING.md#option

File: exporters/otlp/otlpmetric/otlpmetricgrpc/go.mod [hunk 8]
```diff
@@ -0,0 +1,74 @@
+module go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc
+
+go 1.14
```
Comment (MrAlias) [inline exporters/otlp/otlpmetric/otlpmetricgrpc/go.mod:3]:
```suggestion
go 1.15
```

File: exporters/otlp/otlpmetric/otlpmetricgrpc/go.mod [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+replace go.opentelemetry.io/otel/exporters/otlp => ../..
+
+replace go.opentelemetry.io/otel/exporters/otlp/otlpmetric => ../
+
+replace go.opentelemetry.io/otel/metric => ../../../../metric
+
+replace go.opentelemetry.io/otel/oteltest => ../../../../oteltest
+
+replace go.opentelemetry.io/otel/trace => ../../../../trace
```
Comment (MrAlias) [inline exporters/otlp/otlpmetric/otlpmetricgrpc/go.mod:22]: Can we unify these below?

File: exporters/otlp/otlpmetric/options.go (hunk 7 above)
Comment (paivagustavo) [reply exporters/otlp/otlpmetric/options.go]: Updated that to conform the project pattern.

**Review by MrAlias — APPROVED** [review]

**Pull request merged**